pandas/core/arrays/categorical.py

```diff
@@ -520,6 +520,9 @@ def astype(self, dtype: Dtype, copy: bool = True) -> ArrayLike:
             if dtype == self.dtype:
                 return self
             return self._set_dtype(dtype)
+        if is_integer_dtype(dtype) and self.isin([np.nan, -np.inf, np.inf]).any():
```

Comment:
`np.isfinite` might be better to use here.

Reply:
I'm going to revert back to using `isin` because `np.isfinite` breaks with input like `Series(["1", "2", "3"], dtype="category")`

Reply:
May also need a `is_float_dtype(self.categories.dtype)`

Reply:
Wouldn't we want it to raise regardless of the dtype of the categories?  E.g., `astype(int)` should raise for this object even though the categories are integers:
```
[1, 2, NaN]
Categories (2, int64): [1, 2]
```

Reply:
@TomAugspurger Do we also need `not is_extension_array_dtype(dtype)` here to rule out nullable integers?

Reply:
It looks like we follow this up with by passing `self` to `np.array`, which would lose the IntegerArray anyway.